lib/tlscontext.c

```diff
@@ -911,7 +908,7 @@ gboolean
 tls_context_set_verify_mode_by_name(TLSContext *self, const gchar *mode_str)
 {
   if (strcasecmp(mode_str, "none") == 0)
-    self->verify_mode = TVM_NONE;
+    self->verify_mode = TVM_UNTRUSTED;
```

Comment:
I think you'll need `TVM_OPTIONAL | TVM_UNTRUSTED` as `TVM_UNTRUSTED` alone won't match due to the bitwise OR in the switch-case in `tls_context_setup_verify_mode`.

Alternatively, just move the `strcmp( .., "none")` line to the line.
```
else if (strcasecmp(mode_str, "optional-untrusted") == 0 || strcasecmp(mode_str, "optional_untrusted") == 0)
```

Reply:
Thank You for your comment. I agree with you and I will move it to the proposed place. (Small note: the switch statement also come to my mind, but the code will not reach it.)

Reply:
Thank You @gaborznagy. I successfully overseen this part twice.